tsdb/head.go

```diff
@@ -1706,17 +1706,27 @@ func (m seriesHashmap) set(hash uint64, s *memSeries) {
 }
 
 func (m seriesHashmap) del(hash uint64, lset labels.Labels) {
-	var rem []*memSeries
-	for _, s := range m[hash] {
-		if !labels.Equal(s.lset, lset) {
-			rem = append(rem, s)
+	for index, s := range m[hash] {
+		if labels.Equal(s.lset, lset) {
```

Comment:
wdyt?
early returns and also reduce cyclomatic complexity
```suggestion
		if !labels.Equal(s.lset, lset) {
		    continue
		}
```

Reply:
thanks